packages/pg/test/integration/client/cancel-tests.js

```diff
@@ -0,0 +1,25 @@
+"use strict"
+var helper = require('./../test-helper')
+var assert = require('assert')
+
+const suite = new helper.Suite()
+
+suite.testAsync('All queries should return a result array', async () => {
+  const pool = new helper.pg.Pool()
+  const client = await pool.connect()
+  const cancledPromise = client.query('SELECT pg_sleep(100000)')
+    .then(() => {
+      throw new Error('this should not resolve because query is cancled')
+    })
+    .catch(err => {
+      return err
+    })
+  await client.cancelActiveQuery()
+  const err = await cancledPromise
+  assert(err instanceof Error)
```

Comment:
This will be true even in the `throw new Error('this should not resolve` case. One option:

```js
.then(
  () => {
    throw new Error('this should not resolve because query is cancelled')
  },
  err => err
)
```

Reply:
oh yeah your right